src/backend/tasks_io/datafeeds/parsers/fms_api/fms_api_match_parser.py

```diff
@@ -121,9 +120,19 @@ def parse(
                     key=lambda team: team["station"],
                 )
             )
+
+            null_team = any(t["teamNumber"] is None for t in sorted_teams)
+            if (
+                null_team
+                and match["scoreRedFinal"] is None
+                and match["scoreBlueFinal"] is None
+            ):
+                continue
+
             for team in sorted_teams:
                 if team["teamNumber"] is None:
-                    null_team = True
+                    continue
```

Comment:
...which allows us to simply skip over individual teams with a `None` team number (for cases where the whole list has a mix of `None` and non-`None` teams)